pkg/reconciler/revision/revision_test.go

```diff
@@ -488,15 +488,78 @@ func TestNoQueueSidecarImageUpdateFail(t *testing.T) {
 	}
 }
 
-func TestGlobalResyncOnConfigMapUpdateRevision(t *testing.T) {
+func TestGlobalResyncOnDefaultCMChange(t *testing.T) {
 	ctx, informers, ctrl, watcher := newTestControllerWithConfig(t, nil)
 
 	ctx, cancel := context.WithCancel(ctx)
 	grp := errgroup.Group{}
 
-	servingClient := fakeservingclient.Get(ctx)
+	rev := testRevision(getPodSpec())
+	revClient := fakeservingclient.Get(ctx).ServingV1().Revisions(rev.Namespace)
+
+	waitInformers, err := controller.RunInformers(ctx.Done(), informers...)
+	if err != nil {
+		t.Fatal("Failed to start informers:", err)
+	}
+	defer func() {
+		cancel()
+		if err := grp.Wait(); err != nil {
+			t.Errorf("Wait() = %v", err)
+		}
+		waitInformers()
+	}()
+
+	if err := watcher.Start(ctx.Done()); err != nil {
+		t.Fatal("Failed to start watcher:", err)
+	}
+
+	grp.Go(func() error { return ctrl.Run(1, ctx.Done()) })
+
+	revClient.Create(rev)
+	revL := fakerevisioninformer.Get(ctx).Lister()
+	if err := wait.PollImmediate(10*time.Millisecond, 5*time.Second, func() (bool, error) {
+		// The only error we're getting in the test reasonably is NotFound.
+		r, _ := revL.Revisions(rev.Namespace).Get(rev.Name)
+		return r != nil && r.Status.ObservedGeneration == r.Generation, nil
+	}); err != nil {
+		t.Fatal("Failed to see Revision reconciliation:", err)
+	}
+	t.Log("Saw revision reconciliation")
+
+	// Re-get it and nillify the CC, to ensure defaulting
+	// happens as expected.
+	rev, _ = revL.Revisions(rev.Namespace).Get(rev.Name)
+	rev.Spec.ContainerConcurrency = nil
+	rev.Generation++
+	revClient.Update(rev)
```

Comment:
This seems unnecessary, why not just nil this out on 498 before calling Create at all?

Reply:
During create it will default to old value (0). This checks that Global resync takes place, not that we just use the default cm.